### PR TITLE
Feat: add unsupported bodyshape toast for wearables

### LINKED ISSUE #2973: Lack of feedback for the user, when equipping wearables that don't match body shape
Users can’t see the wearables in the backpack when their avatar’s body shape doesn’t match the one of the wearable. 
This is pretty confusing, as the users don’t have a visual feedback on why they’re not able to see them. 
Once proposed solution could be to disabled or gray them out.

Acceptance Criteria

- [ ] Unavailable wearables must be shown with its thumbnail greyed out.
- [ ] When the item is focused a toast will appear explaining why it can't be used. `Wearable not compatible with current body shape.`

[Figma link](https://www.figma.com/file/HEjNjFOQUcCgLK3lDMFrw5/DCL-Design-System-%26-Guidelines?node-id=1585%3A1695)

### PR DESCRIPTION
**Pull request by davidejensen**
## What does this PR change?

Fix #2973 
Adds a small toast to give feedback on unusable wearables due to bodyshape

...

## How to test the changes?

1. Go to: https://play.decentraland.zone/?renderer-branch=feat/wearable-feedback
2. Open the backpack
3. Based on the body shape verify which wearables are grayed out and which not
4. Verify that hovering with the mouse shows the correct toast

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

**Review by kwspgoliasz — CHANGES_REQUESTED** [review]
The feature appears to be implemented mostly correctly, however, a single issue with incorrectly marked section was found. When entering the backpack's body section, the unused avatar type is greyed out and marked as not compatible. Moreover, the icon color for the newly chosen avatar is regained only after switching tabs - https://user-images.githubusercontent.com/110464313/191733604-53607185-4f14-4c22-aaa1-8135cb452076.mp4.

Additionally, it might also be worth to grey out the entire item square and not only the item itself, as some wearables, especially black, white and grey ones, may visually be mistaken as available, despite being limited only to the other avatar type.

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/AvatarEditorHUD/Scripts/ItemToggle.cs [hunk 1]
```diff
@@ -84,8 +87,11 @@ public virtual void Initialize(WearableItem w, bool isSelected, int amount, NFTI
 
     public void OnPointerEnter(PointerEventData eventData)
     {
-        if (!ShowReplacementWarningPanel())
-            ShowHidingWarningPanel();
+        if(!ShowIncompatibleWarningPanel())
+        {
+            if (!ShowReplacementWarningPanel())
```
Comment (Kinerius) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/AvatarEditorHUD/Scripts/ItemToggle.cs:92]: This IF chain getting a bit hard to follow, I would refactor this just for clarity.

Something like
```C#
if (IsIncompatible())
{
    ShowIncompatibleWarning();
}
else if (IsGoingToReplaceSomething())
{
    ShowReplacementPanel();
}
else
{
    ShowHidingWarningPanel();
}
```
 
bool IsGoingToReplaceSomething

**Review by Kinerius — CHANGES_REQUESTED** [review]

**Review by kwspgoliasz — APPROVED** [review]
The feature is implemented correctly and all of the issues reported to the owner of the PR were fixed.

**Review by Kinerius — APPROVED** [review]
LGTM! :D
